portfolio/src/main/webapp/script.js

```diff
@@ -33,23 +68,35 @@ function gotoRandomSite() {
   window.location.href = site;
 }
 
+/**
+ * Gets comments from datastore and stores them in Comment objects.
+ * Also updates the DOM with new comments.
+ */
 function getComments() {
   // Perform the fetch and store as promise
   var commentsPromise = fetch('/data');
   commentsPromise.then(response => response.json())
     .then((resJson) => {
-      comments = resJson;
+      for (const comment of resJson) {
+        commentData.push(new Comment(comment['username'],
```

Comment:
I think this can also be achieved with `commentData.push(Object.assign(new Comment(), comment));`
This way a change in set of fields in the class will not require changes here.

Reply:
This was an awesome tip! Thank you! I had to make sure the field names match in Java and in the JS, but that was a small fix for a big difference